src/mappings/horizonStaking.ts

```diff
@@ -0,0 +1,78 @@
+import { BigInt } from '@graphprotocol/graph-ts'
+import { addresses } from '../../config/addresses.template'
+import { HorizonStakeDeposited, HorizonStakeLocked, HorizonStakeWithdrawn} from '../types/HorizonStaking/HorizonStaking'
+import { Indexer } from '../types/schema'
+import { calculateCapacities, createOrLoadEpoch, createOrLoadGraphNetwork, createOrLoadIndexer, updateAdvancedIndexerMetrics } from './helpers/helpers'
+
+
+export function handleHorizonStakeDeposited(event: HorizonStakeDeposited): void {
+    // To Do
+    let graphNetwork = createOrLoadGraphNetwork(event.block.number, event.address)
+    // update indexer
+    let indexer = createOrLoadIndexer(event.params.serviceProvider, event.block.timestamp)
+    let previousStake = indexer.stakedTokens
+    indexer.stakedTokens = indexer.stakedTokens.plus(event.params.tokens)
+    indexer = updateAdvancedIndexerMetrics(indexer as Indexer)
```

Comment:
For the advanced metrics we have stake ratios and reward/fees cuts. These are now per data service but I assume we will keep them in the `Indexer` entity for stage 1 and assume its for `SubrgaphService` correct?

If so, there are some adjustments we need to make. Not sure this PR is where we should do it, just want to make sure it doesn't slip:
- For stake ratios, we have to consider that they should be calculated with stake that is provisioned/delegated to the `SubgraphService`, not ALL stake should count.
- For cuts, we need to consider that they work slightly different now (they mean the opposite) and that the old values wont carry over to the new ones (indexers need to set the cuts on the `SubgraphService`).

Reply:
Yup, all of that is going to be part of the Provision configuration PR (where we actually set the new delegation cuts), which I'm implementing today 😅